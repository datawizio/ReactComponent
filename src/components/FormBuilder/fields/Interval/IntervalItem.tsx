import React, { useContext } from "react";
import { IntervalItemProps } from "../../types";
import DatePicker from "../../../DatePicker";
import ConfigContext from "../../../ConfigProvider/context";

export const IntervalItem: React.FC<IntervalItemProps> = ({
  label,
  value,
  format,
  picker,
  onChange
}) => {
  const { translate } = useContext(ConfigContext);
  return (
    <>
      <div className="field-interval-label">{label}:</div>
      <DatePicker
        picker={picker as any}
        placeholder={translate("UNLIMITED")}
        value={value}
        //@ts-ignore
        onChange={onChange}
        format={format}
      />
    </>
  );
};

IntervalItem.defaultProps = {
  picker: "month"
};
