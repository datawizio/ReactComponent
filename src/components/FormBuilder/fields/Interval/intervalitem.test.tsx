import "jsdom-global/register";
import React from "react";
import { shallow } from "enzyme";

import { IntervalItem } from "./IntervalItem";

const mockProps = {
  label: "label",
  value: "value",
  onChange: jest.fn()
};

const setUp = (props?) => shallow(<IntervalItem {...props} />);

xdescribe("FormBuilder IntervalItem component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("IntervalItem rendered correctly", () => {
    expect(component).toMatchSnapshot();
  });
});
