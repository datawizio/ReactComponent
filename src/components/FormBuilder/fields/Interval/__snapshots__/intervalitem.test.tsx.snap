// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormBuilder IntervalItem component IntervalItem rendered correctly 1`] = `
<Fragment>
  <div
    className="field-interval-label"
  >
    label
    :
  </div>
  <Picker
    disabledDate={[Function]}
    onChange={[MockFunction]}
    picker="month"
    placeholder="UNLIMITED"
    value="value"
  />
</Fragment>
`;
